serpentTools/objects/xsdata.py

```diff
@@ -0,0 +1,116 @@
+        -8  : "Total fission energy production",
+        -9  : "Majorant macro",
+        -10 : "Macro scattering recoil heating",
+        -11 : "Source rate",
+        -15 : "neutron density",
+        -16 : "Macro total scattering neutron production",
+        -53 : "Macro proton production",
+        -54 : "Macro deutron production",
+        -55 : "Macro triton production",
+        -56 : "Macro He-3 production",
+        -57 : "Macro He-4 production",
+        -100: "User response function" }
+
+    def __init__(self, name, metadata, isIso=False):
+        NamedObject.__init__(self, name)
+
+        # Whether this describes individual isotope XS, or whole-material XS
+        # serpent starts their names with an "m" if it's a whole-material XS
+        self.isIso = isIso
+
+        # metadata reference
+        self.metadata = metadata
+
+        # Possible reactions on this material / nuclide
+        # Maps MT integer number to a description
+        self.MT = []
+        self.MTdescrip = []
+
+        # Holds XS numeric data
+        self.xsdata = None
+
+        # whether nu data present for fissionables
+        self.hasNuData = False
+
+    @staticmethod
+    def negativeMTDescription(mt):
+        """ Gives descriptions for negative MT numbers used by Serpent
+        for whole materials, for neutrons only. """
+        if mt > 0:
+            error("Uh, that's not a negative MT.")
+        try:
+            return XSData.MTdescriptions[mt]
+        except KeyError:
+            error("Cannot find description for MT {}.".format(mt))
+            return None
+
+    def setMTs(self, chunk):
+        """ Parse chunk to MT numbers and descriptions"""
+        if not self.isIso:
+            self.MT = [int(c) for c in chunk[1:]]
+        else:
+            self.MT = [int(c.split('%')[0]) for c in chunk[1:]]
+
+        # Make MT descriptions
+        if not self.isIso:
+            for mt in self.MT:
+                self.MTdescrip.append(self.negativeMTDescription(mt))
+        else:
+            self.MTdescrip = [c.split('%')[1].rstrip() for c in chunk[1:]]
+
+    def setData(self, chunk):
+        """ Parse data from chunk to np array."""
+        self.xsdata = np.zeros([len(self.metadata['egrid']), len(self.MT)])
+        for i, line in enumerate(chunk[1:]):
+            self.xsdata[i,:] = np.array(line.split(), dtype=np.float64)
+
+    def setNuData(self, chunk):
+        """ Add fission neutrons per fission data """
+        self.hasNuData = True
+        self.nuData = np.zeros([len(self.metadata['egrid']),2], dtype=np.float64)
+        for i, cc in enumerate(chunk[1:]):
+            self.nuData[i,:] = np.array(cc.split(), dtype=np.float64)
+
+    def hasExpectedData(self):
+        """ Check that the expected data (MT numbers, an energy grid, etc)
+        were collected. """
+
+        if not isinstance(self.xsdata, np.ndarray):
+            return False
+        if not self.MT:
+            return False
+
+        return True
```

Comment:
Any plans to implement plotting?

Reply:
yes, I getting automated plotting in is an absolute must, IMO. I'm also planning to add some automatic LaTeX table generation for few group stuff in reports. Maybe table generation like that would go better in its own serpentTools class though. Hm.

Reply:
LaTeX tabling would be awesome. For generality, we could utilize something like [`pandas.DataFrame.to_latex`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_latex.html) or a [`jinja2` template](http://jinja.pocoo.org/).

I do want to do more with the templating system given with jinja, like for writing materials and maybe cross sections for nodal diffusion. For now, since this is a one-off application, I don't think we need a dedicated class/module for this, not yet at least. But I am open to the idea in the future